src/ChaiVM/interpreter/executor.cpp

```diff
@@ -109,6 +108,86 @@ void Executor::divi(Instruction ins) {
         decoder::parse(codeManager_->getBytecode(regFile_.pc()));
     (this->*handlerArr[newIns.operation])(newIns);
 }
+void Executor::ldiaf(Instruction ins) {
+    double immd = static_cast<double>(std::bit_cast<float>(ins.immidiate));
+    regFile_.acc() = std::bit_cast<chsize_t>(immd);
+    advancePc();
+    Instruction newIns =
```

Comment:
@c71n93 could you please replace these segments of code with function or macros

Reply:
@levBagryansky how would you call this macro or function?

Reply:
@levBagryansky done